vcs/gitcmd/repo.go

```diff
@@ -150,23 +150,55 @@ func (r *Repository) ResolveTag(name string) (vcs.CommitID, error) {
 	return commitID, nil
 }
 
-func (r *Repository) Branches() ([]*vcs.Branch, error) {
+// branchCounts returns the behind/ahead commit counts information for branch, against base branch.
+func (r *Repository) branchCounts(branch, base string) (behind, ahead int, err error) {
+	cmd := exec.Command("git", "rev-list", "--count", "--left-right", base+"..."+branch)
```

Comment:
I don't think we should rely on the input arguments being safe for 2 reasons:

1. Someone might call this method from somewhere else that gets the arguments from user input, and not think about this issue.
2. `git show-ref --heads` might print out a warning message that is in part determined by user input (or the name of a file that an attacker could commit), and we might parse that warning message (it's probably going to stderr, but we might combine both, or it might go to stderr) as a branch name. I can't think of a way right now, but I can't guarantee it wouldn't happen. (E.g., suppose git warned if you had really long branch names that were deprecated and going to be forbidden in git 17.0 in the future. Say the branch name is `aaaaaaaaaaaaaaaaaaaasdf`. The warning message might be `warning: branch name is too long, truncating 'aaaasdf'`. Then if the branch was named like `aaaaaaaaaaaaaaa--foo`, we might parse that warning message as a branch named `--foo`. Totally hypothetical.)

And if it's impossible to create branch names that begin with a dash, then checking the args will never break the functionality on a valid branch name, so there's no downside.